src/rviz/visualization_manager.cpp

```diff
@@ -121,6 +140,10 @@ class VisualizationManagerPrivate
   ros::NodeHandle update_nh_;
   ros::NodeHandle threaded_nh_;
   boost::mutex render_mutex_;
+  int venc_width_,venc_height_;
+  VideoEncoder *venc_,*venc_keyed_;
```

Comment:
Should always be:
`VideoEncoder* venc_ = NULL;` and 
`VideoEncoder* venc_keyed_ = NULL;` as you are explicitly testing for `venc_ == NULL` later. There is no guarantee that this is going to be null if left uninitialized.

Reply:
Ah, great catch. In my C++ noobery, mistakenly assumed that `new` initialized memory to zero.

Reply:
Why would you want that and not the random leftovers from previous applications?  I did this so many times in the initial implementation :)